src/NUnitFramework/framework/Constraints/UniqueItemsConstraint.cs

```diff
@@ -254,18 +254,16 @@ public NUnitStringEqualityComparer(bool ignoreCase)
 
             public bool Equals(string x, string y)
             {
-                string s1 = _ignoreCase ? x.ToLower() : x;
-                string s2 = _ignoreCase ? y.ToLower() : y;
-
-                return s1.Equals(s2);
+                var stringComparison = _ignoreCase ? StringComparison.CurrentCultureIgnoreCase : StringComparison.Ordinal;
+                return x.Equals(y, stringComparison);
             }
 
             public int GetHashCode(string obj)
             {
                 if (obj is null)
                     return 0;
                 else if (_ignoreCase)
-                    return obj.ToLower().GetHashCode();
```

Comment:
Not only does this allocate, I found through testing that this code was incorrect and could erronously put some things which should be the same into different buckets.